uint32                      id
uint32                      step
string                      type